gammapy/estimators/map/core.py

```diff
@@ -809,6 +813,45 @@ def from_stack(cls, maps, axis, meta=None):
             data=data, reference_model=reference.reference_model, meta=meta, gti=gti
         )
 
+    def split_by_axis(self, axis_name):
```

Comment:
I just opened #3899, after this is merged my proposal would be to implement `FluxPoints.iter_by_axis(axis_name, keepdims=)`. Then the implementation becomes something along the lines of:
```python3
datasets = Datasets()

for fp in flux_points.iter_by_axis(axis_name="time", keepdims=True):
    gti = fp.geom.axes["time"].to_gti()
    dataset = FluxPointsDataset(fp, gti=gti)
    datasets.append(datasets)
```
Where `TimeMapAxis.to_gti()` would still need to be implemented. What do you think @AtreyeeS?

Reply:
We might need to distinguish between the time axis bins and the gti, no?
For example, the `time_axis` has a constant binning, say 6 hrs. And within that 6 hrs, the telescope was operational only for 2 hrs. The time integration (that is taken from the gti) in this case should be 2hrs (and that is what we take within MapEvaluator)

Reply:
Yes, you are right. The code example above does not make sense, it would only work for the "flare" case, where we split into smaller GTIs. So the handling of GTIs remains in `FluxPoints.iter_by_axis()`  based on `gti.select_time()` and makes the code example form the user side even simple:
```python3
datasets = Datasets()

for fp in flux_points.iter_by_axis(axis_name="time", keepdims=False):
    dataset = FluxPointsDataset(fp)
    datasets.append(datasets)
```

Reply:
Should I remove `FluxPoints.slice_by_axis()` (but keep `Map.slice_by_axis`)?